topiary-config/src/language.rs

```diff
@@ -90,87 +89,179 @@ impl Language {
     }
 
     #[cfg(not(target_arch = "wasm32"))]
+    // NOTE: Much of the following code is heavily inspired by the `helix-loader` crate with license MPL-2.0.
+    // To be safe, assume any and all of the following code is MLP-2.0 and copyrighted to the Helix project.
     pub fn grammar(&self) -> TopiaryConfigResult<topiary_tree_sitter_facade::Language> {
-        Ok(match self.name.as_str() {
-            #[cfg(feature = "bash")]
-            "bash" => tree_sitter_bash::language(),
+        let mut library_path = crate::project_dirs().cache_dir().to_path_buf();
+        if !library_path.exists() {
+            std::fs::create_dir(&library_path)?;
+        }
 
-            #[cfg(feature = "css")]
-            "css" => tree_sitter_css::language(),
+        library_path.push(self.name.clone());
 
-            #[cfg(feature = "json")]
-            "json" => tree_sitter_json::language(),
+        if !library_path.exists() {
+            std::fs::create_dir(&library_path)?;
+        }
 
-            #[cfg(feature = "nickel")]
-            "nickel" => tree_sitter_nickel::language(),
+        library_path.push(self.config.grammar.rev.clone());
+        // TODO: Windows?
+        library_path.set_extension("so");
```

Comment:
`dll`, I guess...but I have the impression that the build toolchain in Windows is quite different. The CC crate supports Windows, apparently -- MSVC or MinGW (GCC) -- so _maybe_ it can be supported.

Reply:
Testing is an issue. I don't have a windows License, and CI doesn't build for Windows either.